packages/task/src/node/task-server.ts

```diff
@@ -112,6 +126,31 @@ export class TaskServerImpl implements TaskServer, Disposable {
                             problems
                         });
                     }
+                    if (taskConfiguration.isBackground) {
+                        const backgroundTaskStatus = this.toBackground.get(event.taskId);
+
+                        const isBeginsPatternMatch = collector.processLineMatchBegin(event.line);
+                        if (isBeginsPatternMatch) {
+                            backgroundTaskStatus!.isBeginsPatternMatch = true;
+                        }
+
+                        // Get the 'activeOnStart' value of the problem matcher 'background' property
+                        const activeOnStart = collector.problemMatchers[0]!.watching!.activeOnStart;
+
+                        const isEndsPatternMatch = collector.processLineMatchEnd(event.line);
+
+                        // Mark ends pattern as matches, only after begins pattern matches
+                        if (isEndsPatternMatch && backgroundTaskStatus!.isBeginsPatternMatch) {
```

Comment:
if `activeOnStart` is true, it is possible that the background task ends while `isEndsPatternMatch === true` and `isBeginsPatternMatch === false`

Reply:
fixed, done !